src/components/hook/useTimeTravel.js

```diff
@@ -0,0 +1,33 @@
+import { useState, useEffect } from "react";
+import React from "react";
+
+export default function useTimeTravel() {
+  const [date, setDate] = useState([]);
+  const [index, setIndex] = useState(null);
+  const [current, setCurrent] = useState("");
+
+  useEffect(() => {
+    if (index !== null) {
+      setCurrent(date[index]);
+    }
+  }, [index]);
+
+  const save = (event) => {
+    setCurrent(event.target.value);
+    setDate((prevState) => [...prevState, event.target.value]);
```

Comment:
the setDate should also be able to insert a new date in between other dates, not just add a new date to the end of the array.